util/topsql/reporter/single_target.go

```diff
@@ -63,19 +63,23 @@ func NewSingleTargetDataSink(registerer DataSinkRegisterer) *SingleTargetDataSin
 		registered: atomic.NewBool(false),
 		registerer: registerer,
 	}
+	return dataSink
+}
 
+// Start starts to run SingleTargetDataSink.
+func (ds *SingleTargetDataSink) Start() {
 	addr := config.GetGlobalConfig().TopSQL.ReceiverAddress
 	if addr != "" {
-		dataSink.curRPCAddr = addr
-		if err := registerer.Register(dataSink); err != nil {
+		ds.curRPCAddr = addr
```

Comment:
Seems that when address is empty at that time, register will not be called. Will register be called later when address becomes not empty?

Reply:
Yes, it will be called later in `trySwitchRegistration`.